### PR TITLE
Pr man sens

### PR DESCRIPTION
**Pull request by tumbili**
@kd0aij I added a scale parameters to adjust the throws in full manual for fixed wings.
Do you have any comments on that?

Fixes: #4598

### CODE REVIEW

**Comment by AndreasAntener** [comment]
Cool! I updated the description to include another issue that will be fixed with this.

**Comment by julianoes** [comment]
@tumbili Wouldn't it make more sense to have limit params for control surfaces which then need to be taken into account by both manual and auto modes? The "throw" of a servo doesn't change because it's manual or not, right?

**Comment by tumbili** [comment]
@julianoes I'm not sure if limiting the servo throws in stabilized modes makes any sense, the only thing you are doing is possibly introducing actuator saturation. If the system has a set of stable gains then I don't think that we need to limit the throws.

**Comment by kd0aij** [comment]
The new parameter is equivalent to what is often called the "travel" adjustment in RC transmitters. It is the parameter that changes when you move the "rate" switch (typically 3-pos) on the TX. So to completely duplicate standard funtionality, there should be 3 values corresponding to high, medium and low rates.
In addition, RC TXs provide "expo" which I assume is understood and "curves" which are necessary for setting up collective pitch rotorcraft throttle and collective pitch channels. (I use expo right now for FPV racers in ACRO mode, and I'll need "curves" for my collective pitch quad)
The "limit param" which @julianoes suggests is normally handled on the "mixer" end, and will hopefully be addressed by making it easier to adjust mixer parameters. I believe Matthew Coleman has been looking at that.

**Comment by julianoes** [comment]
@tumbili: I don't mean "limit" in a sense to get actuator saturation but rather you just rescale it to the available output.

@kd0aij: And yes it is true that the mixer does the same that I'm suggesting.

So if I understand it right, you want to have less travel in manual than you have in auto modes. I would argue that you could just use expo on the Tx for the same feel. Is that not enough?

**Comment by kd0aij** [comment]
"feel" and "enough" are non-scientific terms which I refuse to use till I've had a few beers :)
In concrete terms, the proposed parameter provides less functionality than is available in typical RC TXs, and the goal (if I understand correctly) is to eliminate all need for programming the TX.

I had suggested earlier that one could accomplish all of this by programming "flight modes" using OpenTX, but that was deemed undesirable.

**Comment by tumbili** [comment]
@kd0aij If the goal is to eliminate the need of programming the TX then this PR is not sufficient. But I recall that you were suggesting to use the expo on the TX.

**Comment by julianoes** [comment]
@kd0aij If the goal is to eliminate Tx functionality, the PR is in the right direction.

Stepping back, what are we trying to achieve by this? According to #2307 it is about the easiest/safest manual handling of an aircraft in an emergency. I'm assuming, we all want to move to more autonomy with as little manual intervention as possible.

Reading from what @tumbili wrote, there are 2 cases in that manual emergency:
1. The control surface throw is too much and it's hard (twitchy/fast) to pilot in manual.
2. The control surface throw is not enough in certain situations. E.g. we have a stall and the pilot takes over in manual and needs all throw to rescue the plane.

This would suggest to me that this PR resolves 1. while preventing 2. if the params are set and by default we have case 2. but can't have 1.

At this point I'll argue, just use expo and you have both cases covered.

**Comment by kd0aij** [comment]
I don't understand how case (2) is covered, if the proposed parameter limits manual throw.

**Comment by julianoes** [comment]
@kd0aij That's what I'm saying. By limiting the manual throw case 2 is not covered. That's why I think that it's not a good idea to limit manual throw in the way this PR suggests.

**Comment by kd0aij** [comment]
Oh, now I understand your point about expo covering both. 
Right now, I can program my OpenTX  to behave differently in "manual" mode and assign multiple "rates", each with its own amount of expo. That would match the way my non-autopilot planes are set up.
But that requires no new parameters, instead the pilot must program his TX.

**Comment by kd0aij** [comment]
On adding "ACRO for FW": (assuming this new "acro" is not rate-controlled as in MC)


Navstate  |    MC Function   |    FW Function         |        Assist Switch    |   Acro Switch  |  Ratt Switch
 ---            |          ---              |           ---                   |           ---                   |       ---             | ---|
 Manual     |    N/A        |             manual low rate |   off         |               off     |            N/A|
Acro        |      rate control      |   manual max rate     |                  off       |                 on      |             off
Rattitude  |     rate/angle       |    N/A           |                    off     |                   on   |                on
Stabilized   |   angle control |    angle control     |             on          |              N/A         |       N/A

The last 3 columns are the multi-channel FM selection switches. It's beginning to look like a "rate" switch would make this look cleaner 

The single-channel selection model with an arbitrary set of 6 (or more) navstates plus mixing in the TX would eliminate the (sometimes false) impression of a hierarchical organization of flight modes. Perhaps it would be better if the multi-channel flight selection more closely reflected the actual set of flags in vehicle_control_mode_s?

**Comment by julianoes** [comment]
I would not re-use `acro` for `manual` because it will lead to confusion. I think it's better to just have the states `Manual` and `Manual low rate` because it describes what it actually is.

Having said that, `acro` should be `rate control` ;).

**Comment by AndreasAntener** [comment]
https://docs.google.com/document/d/1pi7YMKu9dhcq6HRAXQ_gMesn2gGrRa7cdNTYLEsNDjE/edit#

**Comment by julianoes** [comment]
@AndreasAntener thanks

**Comment by tumbili** [comment]
@AndreasAntener @julianoes @kd0aij What do we do with this one? Is it good enough for now?

File: src/modules/fw_att_control/fw_att_control_params.c [hunk 1]
```diff
@@ -502,3 +502,46 @@ PARAM_DEFINE_FLOAT(FW_FLAPS_SCL, 1.0f);
...
+ * Scale factor applied to the desired yaw actuator command in full manual mode. This parameter allows
+ * to adjust the throws of the control surfaces.
+ *
+ * @unit norm
+ * @min 0.0
+ * @decimal 2
+ * @increment 0.01
+ * @group FW Attitude Control
+ */
+PARAM_DEFINE_FLOAT(FW_MAN_Y_SC, 1.0f);
```
Comment (AndreasAntener) [inline src/modules/fw_att_control/fw_att_control_params.c]: new line missing

**Comment by AndreasAntener** [comment]
Yes, I would say so, the discussion diverted a bit, but basically I think it's a start thinking about integrating it in the new flight mode concept.

**Comment by tumbili** [comment]
@kd0aij Could you give this a try by chance? It would also be important to test the trim functionality via QGC.

**Comment by kd0aij** [comment]
How is trim supposed to work, and what is the test procedure?

**Comment by tumbili** [comment]
@kd0aij In manual mode you use the trim of your RC to get the desired deflection (usually this would be done while you are flying). Then you go into QGC and press 'Copy Trims' under the RC tab.
Then you center the trims on the RC again and you should notice that the deflection of the control surface still reflects the desired trim.

**Comment by kd0aij** [comment]
@tumbili I see, and now I realize why I've never used the trim parameters: 
This mechanism is intended to substitute for "flight mode specific" trims in the TX. They need to be mode specific because in fly-by-wire modes, the "trim" value must always be zero. So if you trim your plane out in manual mode, you will find that things are screwed up in all other modes, because of the non-zero trims.
But doesn't it seem too inconvenient to have to go to the ground station and execute "Copy Trims" every time you retrim in manual (after all, you're doing this in flight)? I'm not sure why anyone with OpenTX would want this feature. What kind of RC controller are we designing for? Perhaps we need a spec for a minimum-capability TX?

**Comment by tumbili** [comment]
@kd0aij I don't see why the trims should only be specific to manual mode. In my opinion the trims are there to find the throws of the control surfaces with which the plane flies more or less neutral. This configuration should also be the starting point for all other modes.
Maybe I'm not getting the point though.

**Comment by kd0aij** [comment]
In manual mode, the manual setpoints (+ trim) are fed directly to the
actuators, whereas in all other modes, they are treated as either angle or
rate setpoints. So the units aren't even the same, e.g. if I need up
elevator trim in manual mode, that will translate to upward pitch angle in
stabilize mode, but certainly not the same amount of up elevator.

**Comment by AndreasAntener** [comment]
@kd0aij the trims are always added directly to the actuator outputs, independent of the mode, it always has been this way. I'm not sure where you see a setpoint trim (yes that would be awkward), it's a bit in transparent maybe in the FW attitude controller.

**Comment by AndreasAntener** [comment]
@kd0aij And you do realize that what you said about flight mode dependent trims is exactly the problem with RC trims?

**Comment by kd0aij** [comment]
Then I was definitely confused. Perhaps I was confusing these parameters with others which specify cruise "trim" orientation?

**Comment by AndreasAntener** [comment]
There are also roll and pitch setpoint offsets, that usually confuses me ;)

**Comment by AndreasAntener** [comment]
Fixed and bench tested with a pure FW setup (IO) and a VTOL setup (FMU).

**Pull request merged**